### DISCUSSION TITLE
Enhancing Laravel: Introducing Enum Translation for a Seamless Developer Experience

### DISCUSSION THREAD

**Discussion by domthomas-dev** [laravel/framework#49464] [Ideas]
Hey Laravel Core Team,

I've got a suggestion that could level up the developer experience with the framework: how about adding a feature to translate enums? Ever thought about it?

Enums are super handy, but things get a bit tricky when we're dealing with different languages. Having a function to translate enums directly would be a game-changer!

Just imagine being able to set translations for each enum value in language files. It would make managing multilingual apps a breeze without making the code overly complicated.

It would be awesome if you could consider this idea for upcoming Laravel versions. It would make developers' lives so much easier and amp up the framework's adaptability.

I'm totally up for a more detailed discussion if you're interested! Thanks for everything you do to evolve Laravel.

I can work on it if you think it's a good idea.

Thanks
Dom

**Comment by domthomas-dev** [comment]
Something like this
https://gist.github.com/domthomas-dev/78e1db6da82512b716b0f8cac6dd93e7#file-localizedenum-php

**Comment by domthomas-dev** [comment]
I don't have any answers, so I created a package, but it would be better directly in the chore. Of course, I can take care of it, with the documentation etc.

https://github.com/domthomas-dev/laravel-localized-enum

**Comment by browner12** [comment]
this is already easy enough to do in userland:

```php
enum Test: string
{
    case FOO = 'foo';
    case BAR = 'bar';
    case BAZ = 'baz';

    public function display(): string
    {
        return match ($this) {
            self::FOO => __('enums/foo'),
            self::BAR => __('enums/bar'),
            self::BAZ => __('enums/baz'),
        };
    }
```

**Reply by domthomas-dev** [reply]
I think it's not really the same. Of course, it's possible to do that in many ways... I find my solution quite interesting, at least for a package, and in the worst case... I use it myself.

**Comment by shaedrich** [comment]
I don't believe, enum values _should_ be used for this (directly).

There's also [`iteks/laravel-enum`](https://github.com/iteks/laravel-enum) to achieve something similar to @browner12's comment with attributes.

**Comment by osama-98** [comment]
I have created a package that manages Enum translation in an elegant way
[https://github.com/osama-98/laravel-enum-translatable](https://github.com/osama-98/laravel-enum-translatable)
